enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
@rhuss given the complexity and number of independently moving parts its kinda hard to just test fmp or kube-client in isolation and to have real confidence its not gonna break something downstream. e.g. the classloader issue you tried fixing is almost impossible to guess ahead of time; its CI that catches it really.

So having downstream CI which tries to recreate & use the fabric8-platform is the best I can think of. e.g. we really need to deploy the platform using new quickstarts using the new fmp versions and client versions to check there's not an issue there, in the pipeline or in the deployment. ClassLoader or version mismatch issues can crop in anywhere really. 

The easiest way to do that is actually to do lots of releases then use fabric8-platform as a CI test before it can release a tested set of dependencies. I'd be totally happy if we can get CI to catch these issues so if fmp/client do a bad release, we can raise an issue and get a downstream release out soon-ish (where we revert the new fmp/client version as a temporary workaround until the real upstream fix is figured out - hopefully automatically too with nice upstream + downstream feedback).

Though maybe another bit of CI we could try do is using the latest fmp to build a new fabric8-platform distribution and try test it out and see if it works? That would have caught this particular issue at least (though not lots of the other ones).

Once we've automated the downstream fabric8-platform CI tests we should be able to reuse that same CI logic for a snapshot test of fmp upstream hopefully. But given the failures that have happened over the last couple of years I'd expect the downstream CI to catch way more issues - but yeah, the earlier we catch things the better